compiler/rustc_mir/src/interpret/eval_context.rs

```diff
@@ -927,7 +931,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     #[must_use]
     pub fn generate_stacktrace(&self) -> Vec<FrameInfo<'tcx>> {
         let mut frames = Vec::new();
-        for frame in self.stack().iter().rev() {
+        for frame in self
+            .stack()
+            .iter()
+            .rev()
+            .skip_while(|frame| frame.instance.def.requires_caller_location(*self.tcx))
```

Comment:
this will also skip stack frames that come *before* the last non-track_caller frame. This is ok, but please add a test for it

Reply:
I used skip_while instead of filter to avoid that. Or do you mean something else?

Reply:
oof... sorry, I completely misprocessed that.